airbyte-protocol/protocol-models/src/main/java/io/airbyte/protocol/models/JsonSchemaPrimitiveUtil.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.protocol.models;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.ImmutableBiMap;
+import com.google.common.collect.ImmutableSet;
+import java.util.Set;
+
+public class JsonSchemaPrimitiveUtil {
+
+  public enum JsonSchemaPrimitive {
+
+    STRING,
+    NUMBER,
+    OBJECT,
+    ARRAY,
+    BOOLEAN,
+    NULL,
+    // V1 schema primitives
+    STRING_V1,
```

Comment:
it makes me a little sad that we need these, but it does seem like the easiest thing to do. Would be cool if we could do `record JsonSchemaPrimitive(Type type, String version)` or something, but maybe that's a really messy refactor?

Reply:
Yeah I'm inclined to do this for now to make the refactor a little manageable. I do like the idea of adding a `record JsonSchemaPrimitive(Type type, String version)`. I think this might make more sense when we add newer protocol versions. 

The other issue is that types may not clearly map 1:1 across protocol versions (I think there was a similar discussion in your PR)

Reply:
yeah, that's fair. Seems good for now and we can always revisit the next time we make big changes to the type system